concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Types.hs

```diff
@@ -69,13 +74,21 @@ genFinalizerRounds =
   where
     genRoundFS = do
         r <- Round <$> arbitrary
+        e <- arbitrary
         fs <- genFinalizerSet
-        return (r, fs)
+        return ((r, e), fs)
+
+-- |Generate round and epoch
+genRoundAndEpoch :: Gen (Round, Epoch)
+genRoundAndEpoch = do
+    r <- Round <$> arbitrary
+    e <- arbitrary
+    return (r, e)
```

Comment:
If there's no correlation between the generated round and epoch, then I think there's no real point generating them as a pair.